jetty-security/src/main/java/org/eclipse/jetty/security/authentication/ClientCertAuthenticator.java

```diff
@@ -131,10 +87,7 @@ public Authentication validateRequest(ServletRequest req, ServletResponse res, b
                         principal = cert.getIssuerDN();
                     final String username = principal == null ? "clientcert" : principal.getName();
 
-                    // TODO: investigate if using a raw byte[] is better vs older char[]
-                    final char[] credential = Base64.getEncoder().encodeToString(cert.getSignature()).toCharArray();
-
-                    UserIdentity user = login(username, credential, req);
+                    UserIdentity user = login(username, "", req);
```

Comment:
I don't like much this empty string and I would prefer null or even better not check credentials.
TCK define a user with empty password and we verify certificate we should not have to check credentials

Reply:
The previous code was using the client certificate signature as a credential.  I think changing this might break any users who may be using this.

If the TCK define an empty password, then we need to be able ask if their is a credential, if no, then we can skip the auth, if not then we should do the previous behaviour.    

Alternately, we can try login with null (or "") credential and if that fails, then try login with the signature as a credential

Reply:
I just wonder how this could work for HashLoginService or few others. The client certificate would be store as a password? Sounds so weird and not safe... I would say as long as the client certificate is matching we could consider the user authenticated as the certificate act as an authentication method and this part should be simplify to simply return the UserIdentity without any password check. would sounds more logical for me. But no an expert in this

Reply:
I don't think this is a great way to have done it, but the old code is clear - it uses the principle (or "clientcert") as the username and the signature as the credential and checks them against the login service, so the HashLoginService must have put the signature in as a password.

I am in favour of just deprecating this class and leaving it's behaviour as is.   If nobody has set an authenticator and we have 1 and only 1 SllContextFactory, then we can instantiate a new `ClientCertSslContextFactoryAuthenticator` class that takes the SslContextFactory